### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by fedaipaca**
Task 1: https://learn.epam.com/myLearning/program?groupGuid=dc6c56f0-d451-44e9-b1e3-568251748d14

What was done:
- Project created with create-react-app with typescript template
- Create Hello component using PureComponent and createElement methods for static Hello message
- Create class based Counter component using Component for basic interactivity
- Create search component using functional component
- Application reviewed in browser using react developer tools extension.

image

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,13 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
+      <Hello></Hello>
+      <Counter></Counter>
+      <SearchForm></SearchForm>
```
Comment (inperegelion) [inline src/App.tsx]: [Q] I see all these 3 components do not have a [child prop](https://reactjs.org/docs/composition-vs-inheritance.html#containment) specified. Can we simply use [self-closing tags](https://reactjs.org/docs/introducing-jsx.html#specifying-children-with-jsx)?

File: src/components/Hello.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const Hello: React.FC = () => {
+  return (
+    <div>
+      <h1>Hello, world</h1>
+    </div>
+  );
+};
+
+export default Hello;
```
Comment (inperegelion) [inline src/components/Hello.tsx]: A few words about the file structure. In React projects we usually create a separate folder for each component. Most of the components will have some CSS in future, and it is convenient to keep it in the same folder. Sth like below:

```
components
+-- Counter
|      index.tsx
|      style.scss
+-- Search
|  |      index.tsx
|  +-- SearchButton
|  |      index.tsx
|  |      interfaces.ts
...
```

Comment (fedaipaca) [reply src/components/Hello.tsx]: That was helpful, thanks for the tip. I thought about it but couldn't find an appropriate convention.

File: src/App.tsx (hunk 1 above)
Comment (fedaipaca) [reply src/App.tsx]: Updated as self-closing.

**Pull request merged**